providers/os/connection/device/linux/device_manager.go

```diff
@@ -75,11 +87,221 @@ func (d *LinuxDeviceManager) IdentifyMountTargets(opts map[string]string) ([]*sn
 		partitions = append(partitions, partitionsForDevice...)
 	}
 
-	return partitions, errors.Join(errs...)
+	if err := errors.Join(errs...); err != nil {
+		return partitions, err
+	}
+
+	if opts[SkipAttemptExpandPartitions] == "true" {
+		return partitions, nil
+	}
+
+	return d.attemptExpandPartitions(partitions)
+}
+
+func (d *LinuxDeviceManager) attemptExpandPartitions(partitions []*snapshot.PartitionInfo) ([]*snapshot.PartitionInfo, error) {
+	fstabEntries, err := d.HintFSTypes(partitions)
+	if err != nil {
+		log.Warn().Err(err).Msg("could not find fstab")
+		return partitions, nil
+	}
+	log.Debug().Any("fstab", fstabEntries).
+		Msg("fstab entries found")
+
+	partitions, err = d.MountWithFstab(partitions, fstabEntries)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to mount partitions with fstab")
+		d.UnmountAndClose()
+		return partitions, nil
+	}
+
+	return partitions, nil
+}
+
+func (d *LinuxDeviceManager) HintFSTypes(partitions []*snapshot.PartitionInfo) ([]resources.FstabEntry, error) {
+	for i := range partitions {
+		partition := partitions[i]
+
+		dir, err := d.volumeMounter.MountP(&snapshot.MountPartitionDto{PartitionInfo: partition})
+		if err != nil {
+			continue
+		}
+		defer func() {
+			if err := d.volumeMounter.UmountP(partition); err != nil {
+				log.Warn().Err(err).Str("device", partition.Name).Msg("unable to unmount partition")
+			}
+		}()
+
+		entries, err := d.AttemptFindFstab(dir)
+		if err != nil {
+			return nil, err
+		}
+		if entries != nil {
+			return entries, nil
+		}
+
+		if ok := d.AttemptFindOSTree(dir, partition); ok {
+			return nil, nil
+		}
+
+	}
+
+	return nil, errors.New("fstab not found")
+}
+
+func (d *LinuxDeviceManager) AttemptFindFstab(dir string) ([]resources.FstabEntry, error) {
+	cmd := exec.Command("find", dir, "-type", "f", "-wholename", `*/etc/fstab`)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Error().Err(err).Msg("error searching for fstab")
+		return nil, nil
+	}
+
+	if l := len(strings.Split(string(out), "\n")); l > 2 {
+		log.Debug().Bytes("find", out).Msg("fstab not found, too many results")
+		return nil, nil
+	} else if l < 2 {
+		log.Debug().Bytes("find", out).Msg("fstab not found, no results")
+		return nil, nil
+	}
+
+	mnt, fstab := path.Split(strings.TrimSpace(string(out)))
+	fstabFile, err := afero.ReadFile(
+		fs.NewMountedFs(mnt),
+		path.Base(fstab))
+	if err != nil {
+		log.Error().Err(err).Msg("error reading fstab")
+		return nil, nil
+	}
+
+	return resources.ParseFstab(bytes.NewReader(fstabFile))
+}
+
+func (d *LinuxDeviceManager) AttemptFindOSTree(dir string, partition *snapshot.PartitionInfo) bool {
+	info, err := os.Stat(path.Join(dir, "ostree"))
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
nit: && the 2 checks

Reply:
two checks are there, so if ostree simply doesn't exist this function fails silently, but if it's something else we get to log the error